server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/QueryParser.java

```diff
@@ -0,0 +1,172 @@
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.vault.routes;
+
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.deletionDate;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.deliveryDate;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.hasAttachment;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.originMailboxes;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.recipients;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.sender;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldName.subject;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldParser.BOOLEAN_PARSER;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldParser.MAIL_ADDRESS_PARSER;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldParser.STRING_PARSER;
+import static org.apache.james.webadmin.vault.routes.QueryParser.FieldParser.ZONED_DATE_TIME_PARSER;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.afterOrEquals;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.beforeOrEquals;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.contains;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.containsIgnoreCase;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.equals;
+import static org.apache.james.webadmin.vault.routes.QueryParser.Operator.equalsIgnoreCase;
+
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import javax.mail.internet.AddressException;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.vault.search.Criterion;
+import org.apache.james.vault.search.CriterionFactory;
+import org.apache.james.vault.search.Query;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableTable;
+
+class QueryParser {
+
+    static class QueryParserException extends Exception {
+        QueryParserException(String message) {
+            super(message);
+        }
+    }
+
+    enum FieldName {
```

Comment:
Shouldn't the enums be in SNAKECASE_UPPERCASE for entries?

Reply:
I find it is convenience to having directly a field matched with a field name. fields in camelCase are needed for computation. Having an extra string value wrapped inside is more compilcated, I think.

Reply:
I got the remark in an other PR that enums should be like that... So open question : is there a good practice here to  adopt as a team, or is it "open bar"?

Reply:
I will refactor them!